homeassistant/helpers/update_coordinator.py

```diff
@@ -79,7 +80,7 @@ def __init__(
         self.update_method = update_method
         self.update_interval = update_interval
         self._shutdown_requested = False
-        self.config_entry = config_entries.current_entry.get()
+        self.config_entry: ConfigEntry | None = config_entries.current_entry.get()
```

Comment:
This seems unneeded? `config_entries.current_entry` is already typed?
Thus, in this specific location, the typing is inferred.

Reply:
![CleanShot 2023-09-20 at 19 35 52](https://github.com/home-assistant/core/assets/195327/7440cd26-2a70-4892-aba0-355a16a03563)

Reply:
![image](https://github.com/home-assistant/core/assets/7083755/aac078ef-5aa6-418b-8aef-00025ba54cdb)
euh, looks like a you problem 👀

Reply:
That's what you get for using light theme :P

Reply:
Ok, lol :D Will check, maybe a missing checkbox

Reply:
Ah yea, you have `config_entry` in your subclass @joostlek:

<img width="615" alt="CleanShot 2023-09-20 at 20 00 31@2x" src="https://github.com/home-assistant/core/assets/3989428/6a39fa7a-188e-4ff4-8ea9-a6bb774b2bed">

The Discovergy update coordinator for example does not.

Reply:
> Interesting, PyCharm inferrers it as Any from .get():

That is a pycharm issue, which you should report upstream.

Going to close this PR for that reason.

../Frenck

Reply:
Thanks @joostlek for the help on Discord :) PyCharm seems to be a bit confused when marking the `tests` folder as the test folder.